parsers/US_NEISO.py

```diff
@@ -225,11 +227,11 @@ def fetch_exchange(
                 netFlow=datapoint["Actual"] * multiplier,
                 source=SOURCE,
             )
+        # Append the exchange list to the unmerged list
+        unmerged_exchanges.append(exchanges)
 
-    # This will merge exchanges with the same datetime
-    exchanges = ExchangeList.merge_exchanges([exchanges], logger)
```

Comment:
While this worked this is very error prone and was the reason I decided to add snapshots here in the first place before fixing it.

The ExchangeList (and all EventLists) assumes there are one event per datetime but it's not enforced, if we changed that in the future this would have created a different output.

By creating intermediate lists and then merging that we solve the issue by having unique datetimes in each list and processing everything correctly.

Reply:
agreed, I've found the ExchangeList confusing before

Reply:
I've wanted to change the internal representation of the lists to dicts where the key is the datetime, it would significantly simplify some of the code but it would have completely broken this as it where before...